### ISSUE TITLE
Remove SAPO `IndexPutBackward0` warning filter once `liger-kernel >= 0.8.2` is required

### ISSUE THREAD

**Issue by albertvillanova** [huggingface/trl#6435]
### What

`test_train_loss_types[sapo-True]` (SAPO loss with `use_liger_kernel=True`) emits a benign `UserWarning` in CI:

```
torch/autograd/graph.py: UserWarning: Error detected in IndexPutBackward0. Traceback of forward call that caused the error:
  File ".../liger_kernel/chunked_loss/grpo_loss.py", line 171, in torch_dynamo_resume_in_ppo_loss_fn_at_172
    per_token_loss[positive_advantages_mask] = sapo_loss_fn(...)
```

Example CI run: https://github.com/huggingface/trl/actions/runs/29507534675/job/87652100190

### Root cause

The warning originates in `liger-kernel`, not TRL. Its SAPO branch assigns the per-token loss with boolean-mask indexing (`per_token_loss[mask] = sapo_loss_fn(coef_1[mask], ...)`). That indexing lowers to `aten.nonzero` (data-dependent output shape), which Inductor cannot trace, so `torch.compile` inserts a graph break. AOTAutograd runs the forward metadata collection under `torch.autograd.detect_anomaly(check_nan=False)`, and the in-place `IndexPutBackward` gets flagged, producing the warning. Training itself is correct — the test passes and parameters update.

### Upstream status

Already fixed upstream by replacing the boolean-mask assignment with `torch.where`: https://github.com/linkedin/Liger-Kernel/pull/1274

The fix is **merged but unreleased** — the latest liger-kernel release is `0.8.0` (which predates the fix), and TRL currently requires `liger-kernel>=0.8.0`.

### Workaround in place

We suppress the warning via `filterwarnings` in `pyproject.toml`:

```toml
"ignore:Error detected in IndexPutBackward0:UserWarning",
```

### Action item

Remove the `filterwarnings` entry once `pyproject.toml` requires `liger-kernel >= 0.8.2`, i.e. the first non-excluded release containing #1274. That bump will happen naturally when 0.8.2 ships with #1316 and the `!=0.8.1` exclusion is dropped; the filter removal belongs in that same PR. Do not bump the floor solely to drop this filter.
